airbyte-webapp/src/pages/SettingsPage/pages/DbMigrationPage/components/MigrationComponent.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from "react";
+import { useFetcher } from "rest-hooks";
+import DbMigrationResource, {
+  DbMigrationInfoItem,
+} from "core/resources/DbMigration";
+import useDbMigration from "components/hooks/services/useDbMigrationHook";
+import { useAsyncFn } from "react-use";
+import { Block, Title } from "../../ConnectorsPage/components/PageComponents";
+import { LoadingButton } from "components";
+import { FormattedMessage } from "react-intl";
+import MigrationTableView from "./MigrationTableView";
+
+type DbMigrationComponentProps = {
+  databaseTitle: string;
+  databaseIdentifier: string;
+};
+
+const MigrationComponent: React.FC<DbMigrationComponentProps> = ({
+  databaseTitle,
+  databaseIdentifier,
+}) => {
+  const { info, migrate } = useDbMigration();
+
+  const [migrations, setMigrations] = useState<Array<DbMigrationInfoItem>>([]);
+  const [hasPendingMigration, setHasPendingMigration] = useState(false);
+
+  useEffect(() => {
+    info(databaseIdentifier).then((result) => setMigrations(result.migrations));
+  }, []);
+
+  useEffect(() => {
+    const hasPending = migrations.some(
+      (migration) => migration.migrationState === "Pending"
+    );
+    setHasPendingMigration(hasPending);
+  }, [migrations]);
+
+  const [{ loading: migrating }, onMigrate] = useAsyncFn(async () => {
+    await migrate(databaseIdentifier);
+    const result = await info(databaseIdentifier);
+    setMigrations(result.migrations);
```

Comment:
Same as above - if shape is declared correctly, you do not need to manage state manually as `rest-hooks` lib will handle that. No need to call `setMigrations` if you need help with the lib - let me know and we could discuss how to change this part.